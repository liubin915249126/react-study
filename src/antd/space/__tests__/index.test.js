import React, { useState } from 'react';
import { render, mount } from 'enzyme';
import { act } from 'react-test-renderer';
import Space from '..';
import ConfigProvider from '../../config-provider';
import mountTest from '../../../tests/shared/mountTest';

describe('Space', () => {
  mountTest(Space);

  it('should render width empty children', () => {
    const wrapper = mount(<Space />);

    expect(wrapper.instance()).toBe(null);
  });

  it('should render width ConfigProvider', () => {
    const wrapper = mount(
      <ConfigProvider space={{ size: 'large' }}>
        <Space>
          <span>1</span>
          <span>2</span>
        </Space>
        <Space size="middle">
          <span>1</span>
          <span>2</span>
        </Space>
        <Space size="large">
          <span>1</span>
          <span>2</span>
        </Space>
      </ConfigProvider>,
    );

    expect(render(wrapper)).toMatchSnapshot();
  });

  it('should render width customize size', () => {
    const wrapper = mount(
      <Space size={10}>
        <span>1</span>
        <span>2</span>
      </Space>,
    );

    expect(wrapper.find('div.ant-space-item').at(0).prop('style').marginRight).toBe(10);
    expect(wrapper.find('div.ant-space-item').at(1).prop('style').marginRight).toBeUndefined();
  });

  it('should render vertical space width customize size', () => {
    const wrapper = mount(
      <Space size={10} direction="vertical">
        <span>1</span>
        <span>2</span>
      </Space>,
    );

    expect(wrapper.find('div.ant-space-item').at(0).prop('style').marginBottom).toBe(10);
    expect(wrapper.find('div.ant-space-item').at(1).prop('style').marginBottom).toBeUndefined();
  });

  it('should render correct with children', () => {
    const wrapper = mount(
      <Space>
        text1<span>text1</span>
        <>text3</>
      </Space>,
    );

    expect(render(wrapper)).toMatchSnapshot();
  });

  it('should render with invalidElement', () => {
    const wrapper = mount(
      <Space>
        text1<span>text1</span>
        text1
      </Space>,
    );

    expect(wrapper.find('div.ant-space-item').length).toBe(3);
  });

  it('should be keep store', () => {
    function Demo() {
      const [state, setState] = React.useState(1);

      return (
        <div
          id="demo"
          onClick={() => {
            setState(value => value + 1);
          }}
        >
          {state}
        </div>
      );
    }
    function SpaceDemo() {
      const [visible, setVisible] = useState(true);
      function onChange() {
        setVisible(!visible);
      }
      return (
        <Space>
          {visible && <div>space</div>}
          <Demo />
          <p onClick={onChange}>Three</p>
        </Space>
      );
    }
    const wrapper = mount(<SpaceDemo />);

    expect(wrapper.find('#demo').text()).toBe('1');

    act(() => {
      wrapper.find('#demo').simulate('click');
    });

    expect(wrapper.find('#demo').text()).toBe('2');

    act(() => {
      wrapper.find('p').simulate('click');
    });

    expect(wrapper.find('#demo').text()).toBe('2');
  });

  it('split', () => {
    const wrapper = mount(
      <Space split="-">
        text1<span>text1</span>
        <>text3</>
      </Space>,
    );

    expect(render(wrapper)).toMatchSnapshot();
  });
});
