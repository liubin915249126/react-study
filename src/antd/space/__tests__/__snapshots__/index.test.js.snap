// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should render correct with children 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
>
  <div
    class="ant-space-item"
    style="margin-right:8px"
  >
    text1
  </div>
  <div
    class="ant-space-item"
    style="margin-right:8px"
  >
    <span>
      text1
    </span>
  </div>
  <div
    class="ant-space-item"
  >
    text3
  </div>
</div>
`;

exports[`Space should render width ConfigProvider 1`] = `
Array [
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right:24px"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right:16px"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right:24px"
    >
      <span>
        1
      </span>
    </div>
    <div
      class="ant-space-item"
    >
      <span>
        2
      </span>
    </div>
  </div>,
]
`;

exports[`Space split 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
>
  <div
    class="ant-space-item"
    style="margin-right:4px"
  >
    text1
  </div>
  <span
    class="ant-space-item-split"
    style="margin-right:4px"
  >
    -
  </span>
  <div
    class="ant-space-item"
    style="margin-right:4px"
  >
    <span>
      text1
    </span>
  </div>
  <span
    class="ant-space-item-split"
    style="margin-right:4px"
  >
    -
  </span>
  <div
    class="ant-space-item"
  >
    text3
  </div>
</div>
`;
